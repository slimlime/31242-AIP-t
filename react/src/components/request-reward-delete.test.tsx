import React from "react";
import { shallow, mount } from "enzyme";
import RewardDelete from "./request-reward-delete";

describe("<Iou />", () => {
  it("should render correctly", () => {
    const wrapper = shallow(
      <RewardDelete
        requestID="1"
        rewardID="1"
        refreshTable={() => {
          return "substitute test function";
        }}
        closeModal={() => {
          return "substitute test function";
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should call the delete reward API endpoint when clicked", () => {
    const spy = jest.spyOn(RewardDelete.prototype, "deleteReward");
    const wrapper = mount(
      <RewardDelete
        requestID="1"
        rewardID="1"
        refreshTable={() => {
          return "substitute test function";
        }}
        closeModal={() => {
          return "substitute test function";
        }}
      />
    );
    wrapper.find("#requestItem").simulate("click");
    expect(spy).toHaveBeenCalledTimes(1);
  });
});
