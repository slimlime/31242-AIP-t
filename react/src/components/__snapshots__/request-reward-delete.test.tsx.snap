// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Iou /> should render correctly 1`] = `
<div
  id="requestItem"
  onClick={[Function]}
>
  <DeleteIcon
    id="deleteIcon"
  />
  <WithStyles(ForwardRef(Snackbar))
    anchorOrigin={
      Object {
        "horizontal": "left",
        "vertical": "bottom",
      }
    }
    autoHideDuration={5000}
    message=""
    onClose={[Function]}
    open={false}
  />
</div>
`;
